app/controllers/users_controller.rb

```diff
@@ -1,8 +1,14 @@
 class UsersController < ApplicationController
   def destroy
-    return unless current_user.destroy!
-
+    destroy_user
     flash[:success] = t('sign_up.cancel.success')
     redirect_to root_path
   end
+
+  private
+
+  def destroy_user
+    user = current_user || User.where(confirmation_token: session[:user_confirmation_token]).take
```

Comment:
I think this could be `User.find_by(confirmation_token: session[:user_confirmation_token])` instead of `.take` since there should only be one?